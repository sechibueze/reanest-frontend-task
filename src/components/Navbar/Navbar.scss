@import "../../assets/theme/colors.scss";

#menu-checkbox {
  display: none;
  user-select: none;
}
#menu-checkbox:checked ~ .wrapper .sidebar {
  display: block;
}
.header {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  background-color: $white;
  padding: 1rem;
}
