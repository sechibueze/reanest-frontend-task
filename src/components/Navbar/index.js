import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Fragment } from "react";

import "./Navbar.scss";
const Navbar = () => {
  return (
    <Fragment>
      <input type="checkbox" id="menu-checkbox" className="menu-toggler" />
      <header className="header">
        <label htmlFor="menu-checkbox">
          <FontAwesomeIcon icon={"bars"} size="2x" />
        </label>
      </header>
    </Fragment>
  );
};

export default Navbar;
