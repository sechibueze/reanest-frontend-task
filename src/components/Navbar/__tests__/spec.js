import { shallow } from "enzyme";
import Navbar from "../index";

describe("<Navbar /> renders correctly", () => {
  const component = shallow(<Navbar />);
  const header = component.find(".header");
  const menuToggler = component.find(".menu-toggler");
  it("Navbar matches snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("Navbar has a header", () => {
    expect(header.length).toEqual(1);
  });
  it("Navbar has a pseudo checkbox for opening sidebar", () => {
    expect(menuToggler.length).toEqual(1);
  });
});
