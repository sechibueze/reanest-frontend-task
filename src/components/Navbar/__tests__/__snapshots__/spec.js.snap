// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `ShallowWrapper {}`;

exports[`<Navbar /> renders correctly Navbar matches snapshot 1`] = `ShallowWrapper {}`;
